tests/core/pyspec/eth2spec/test/altair/transition/test_slashing.py

```diff
@@ -0,0 +1,133 @@
+    """
+    # slash 1/4 validators
+    slashed_indices = slash_some_validators_for_inactivity_scores_test(
+        spec, state, rng=random.Random(5566), fraction=0.25)
+    assert len(slashed_indices) > 0
+
+    # check if some validators are slashed but still active
+    for validator_index in slashed_indices:
+        validator = state.validators[validator_index]
+        assert validator.slashed
+        assert spec.is_active_validator(validator, spec.get_current_epoch(state))
+    assert not spec.is_in_inactivity_leak(state)
+
+    transition_until_fork(spec, state, fork_epoch)
+
+    assert spec.get_current_epoch(state) < fork_epoch
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    blocks = []
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch)
+    blocks.append(post_tag(block))
+
+    # ensure that some of the current sync committee members are the slashed
+    slashed_pubkeys = [state.validators[index].pubkey for index in slashed_indices]
+    assert any(set(slashed_pubkeys).intersection(list(state.current_sync_committee.pubkeys)))
+
+    # continue regular state transition with new spec into next epoch
+    to_slot = post_spec.SLOTS_PER_EPOCH + state.slot
+    # since the proposer might have been slashed, here we only create blocks with non-slashed proposers
+    blocks.extend([
+        post_tag(block) for block in
+        state_transition_across_slots_with_ignoring_proposers(post_spec, state, to_slot, slashed_indices)
+    ])
+
+    # check post state
+    for validator in state.validators:
+        assert post_spec.is_active_validator(validator, post_spec.get_current_epoch(state))
+    assert not post_spec.is_in_inactivity_leak(state)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=2)
+@always_bls
+def test_transition_with_attester_slashing_at_fork(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Create an attester slashing at the transition
+    """
+    transition_until_fork(spec, state, fork_epoch)
+
+    yield "pre", state
+
+    # NOTE: it can only be created with pre spec
+    attester_slashing = get_valid_attester_slashing(spec, state, signed_1=True, signed_2=True)
+    operation_dict = {'attester_slashings': [attester_slashing]}
+
+    # irregular state transition to handle fork:
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch, operation_dict=operation_dict)
+    blocks = []
+    blocks.append(post_tag(block))
+
+    indices = set(attester_slashing.attestation_1.attesting_indices).intersection(
+        attester_slashing.attestation_2.attesting_indices
+    )
+    assert len(indices) > 0
+    for validator_index in indices:
+        assert state.validators[validator_index].slashed
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
```

Comment:
this test will be lighter if we only go a few blocks/slots into this epoch, rather than the full `SLOTS_PER_EPOCH` and it seems like additional blocks don't add much value considering what this function aims to test

Reply:
Good call 👍
I changed it to using `only_at(to_slot)` so that it only produces the last block. What do you think?